### ISSUE TITLE
[Feature] Show error for referenced but unassigned variable with default ("${variable:-}")

### ISSUE THREAD

**Issue by lucasbasquerotto** [koalaman/shellcheck#2270]
#### For new checks and feature suggestions
- [X] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

Although not exactly related, what I want is almost the opposite of:

 https://github.com/koalaman/shellcheck/issues/906

#### Here's a snippet or screenshot that shows the problem:

Assuming that `./files/env.sample.sh` has exports for `env1`, `env2`, `env3` and `env4`:

```sh
#!/bin/bash
# shellcheck source=./files/env.sample.sh
source "/path/to/env.sh"

var1="$env1"
var2="${env2:-}"
var3="$env5"
var4="${env6:-}"
```

#### Here's what shellcheck currently says:

Only `env5` shows an error (`env5 is referenced but not assigned - shellcheck(SC2154)`)

#### Here's what I wanted or expected to see:

I would expect errors for `env5` and `env6`. Although I kind of understand it not giving an error because of the default assignment, I see it causing more harm than good (for not reporting), because I frequently use default values for variables, but if some of the variables are misspelled everything would seem to be fine, even tough it isn't. 

This already happened a handful of times, because I use sourced files in several projects, and not all variables need to be present, so I use defaults for them, but expects an error if I wrote the variable name wrong (or if some time later I change its name or remove it from the environment file). 

So, I would expect such a validation to happen, but it doesn't need to be `shellcheck(SC2154)`, actually it's better if it isn't, because if someone wants to disable this validation for variables with defaults, it wouldn't need to disable the already existing rule.
